### PR TITLE
feat: fix uv-trampoline renovate issues

### PR DESCRIPTION
**Pull request by samypr100**
## Summary

1. Fixes errors from https://github.com/astral-sh/uv/pull/4878
2. More cleanup. Removed the need for `MaybeUninit` and `SizeOf` helpers. Renamed main entrypoint to expected default in windows `mainCRTStartup` to avoid re-declaring /ENTRY in build.rs.
3. Adds a small basic test harness that >>on windows<< will generate both types of launchers and run them. I've had been using this locally to test changes and edge cases, but it might be useful for others. It's based on core parts of install-wheel-rs.

## Test Plan

Tested locally on a couple of script/gui apps.

### CODE REVIEW

**Comment by zanieb** [comment]
Thanks for owning this!

File: crates/uv-trampoline/tests/harness.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    // Locate an arbitrary python installation from PATH
+    let python_executable_path = which("python")?;
+    let pythonw_executable_path = which("pythonw")?;
+
+    // Generate Launcher Scripts
+    let launcher_console_script =
+        get_script_launcher(&format_shebang(&python_executable_path), false);
+    let launcher_gui_script = get_script_launcher(&format_shebang(&pythonw_executable_path), true);
+
+    // Genera Launcher Payload
```
Comment (konstin) [inline crates/uv-trampoline/tests/harness.rs:206]:
```suggestion
    // Generate Launcher Payload
```

File: crates/uv-trampoline/tests/harness.rs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            .expect("File Path to be smaller than 4GB")
+            .to_le_bytes(),
+    );
+    launcher.extend_from_slice(&LAUNCHER_MAGIC_NUMBER);
+
+    Ok(launcher)
+}
+
+#[test]
+fn generate_test_launchers() -> Result<()> {
```
Comment (konstin) [inline crates/uv-trampoline/tests/harness.rs:191]: Could you split this into two tests? One that runs the CLI test, and another that needs the user to close the window annotated with `#[ignore]`

**Review by konstin — APPROVED** [review]

File: crates/uv-trampoline/tests/harness.rs (hunk 2 above)
Comment (samypr100) [reply crates/uv-trampoline/tests/harness.rs]: Done in 20df349646b6ac85fb9b7e01098f6018e4d67f76

**Review by konstin — APPROVED** [review]
Thanks!
